File: src/types/repository.ts
export type RepositorySummary = {
  path: string;
  name: string;
  currentBranch?: string;
  headHash?: string;
  isBare: boolean;
  isEmpty: boolean;
  isDetached: boolean;
};

export type RecentRepository = {
  path: string;
  name: string;
  lastOpened: string;
};

export type RepositoryError = {
  code: string;
  message: string;
};

File: src-tauri/src/models/repository.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RepositorySummary {
    pub path: String,
    pub name: String,
    pub current_branch: Option<String>,
    pub head_hash: Option<String>,
    pub is_bare: bool,
    pub is_empty: bool,
    pub is_detached: bool,
}
